xmodule/tabs.py

```
@@ -281,7 +281,7 @@ def fragment_view(self):
         Returns the view that will be used to render the fragment.
         """
         if not self._fragment_view:
-            self._fragment_view = get_storage_class(self.fragment_view_name)()
+            self._fragment_view = import_string(self.fragment_view_name)()
```

The underlying implementation falls back to the default storage class: https://github.com/django/django/blob/stable/4.2.x/django/core/files/storage/__init__.py#L37

Should we do the same here?

---

I don't think so. Here's why: this line of code is meant to import a class, not to use a storage class.
A good example is this line: https://github.com/openedx/edx-platform/blob/master/lms/djangoapps/discussion/plugins.py#L30 — the class there uses the self.fragment_view_name attribute, which refers to a regular view, not a storage class. It’s a bit odd.
I suppose they just used get_storage_class because it internally calls [import_string](https://github.com/django/django/blob/591b23a11b099081fc39d00171f5dc9efbd67e13/django/core/files/storage/__init__.py#L37).

---

That is very weird, it sounds like actually getting a storage class here would be a bug and we're fixing that bug either the view you tried to import is imported and called or an error is thrown.  I'm a little worried that previously this failed silently and now it doesn't which may expose other issues that were previously hidden.  I think this makes sense but I'd consider it a slightly risky change so I'll inform people about this before it is merged.